devs/docs/architecture.rst

```diff
@@ -0,0 +1,98 @@
+=====================
+CrateDB Architecture
+=====================
+
+This document assumes that you are familiar with using CrateDB. If you haven't
+used CrateDB before, please check out the [Getting
+Started](https://crate.io/docs/crate/getting-started/en/latest/) section of the
+user documentation.
+
+CrateDB is a distributed SQL database. From a high-level perspective, the
+following components can be found in each node of CrateDB:
+
+.. image:: ../resources/architecture.svg
+
+Components
+-------------
+
+SQL Engine
+............
+
+CrateDB's heart consists of a SQL Engine which takes care of parsing SQL
+statements and executing them in the cluster.
+
+The engine is comprised of the following components:
+
+1. Parser: Breaks down the SQL statement into its components.
+2. Analyzer: Performs semantic processing of the statement including
+   verification, type annotation, and normalization.
+3. Planner: Builds an execution plan from the analyzed statement and optimizes it if possible.
+4. Executor: Executes the plan in the cluster and collects results.
+
+Class entry point:
+
+- `SqlParser`
+- `Analyzer`
+- `Planner`
+- `LogicalPlan` / `Plan`
+- `NodeOperationTree`
+- `ExecutionPhasesTask`
+- `BatchIterator`
+
+Input
+.....
+
+CrateDB accepts input from an HTTP/REST interface as well as from a Postgres
+compatible protocol. The Admin UI, crash and some of the connectors use the HTTP
+interface whereas the JDBC client uses the Postgres protocol.
+
+An advantage of the Postgres protocol in CrateDB is that you can use it also for
+applications which were originally built to communicate with Postgres. You may
+also use Postgres tools like `psql` with CrateDB.
+
+Class entry points:
+
+- `CrateRestMainAction`
+- `PostgresWireProtocol`
+
+Transport
+..........
+
+Transport denotes the communication between nodes in a CrateDB cluster. CrateDB
+uses Netty and Elasticsearch to transfer data between nodes.
```

Comment:
Might it be worth briefly mentioning what *kinds* of communication occur on the transport layer? Like synchronising the cluster start, redistributing shards, node discovery, etc?

Reply:
👍 I've pushed a fixup.